internal/cmd/upgradecmd.go

```diff
@@ -432,9 +443,16 @@ func (c *Config) verifyChecksum(ctx context.Context, rr *github.RepositoryReleas
 	return nil
 }
 
-// getMethod attempts to determine the method by which chezmoi can be upgraded
-// by looking at how it was installed.
-func getMethod(fileSystem vfs.Stater, executableAbsPath chezmoi.AbsPath) (string, error) {
+// getUpgradeMethod attempts to determine the method by which chezmoi can be
+// upgraded by looking at how it was installed.
+func getUpgradeMethod(fileSystem vfs.Stater, executableAbsPath chezmoi.AbsPath) (string, error) {
+	switch {
+	case runtime.GOOS == "darwin" && strings.Contains(executableAbsPath.String(), "/homebrew/"):
+		return upgradeMethodBrewUpgrade, nil
+	case runtime.GOOS == "linux" && strings.Contains(executableAbsPath.String(), "/.linuxbrew/"):
+		return upgradeMethodBrewUpgrade, nil
```

Comment:
This is risky... someone can install Homebrew wherever they want.

Should not we check the version of chezmoi itself? It says that it was built by Homebrew (`chezmoi --version`).

But to be really bullet proof : 

1. Check if `brew` command exists.
2. Get brew prefix `brew --prefix`
3. Get chezmoi path and check if it contains `brew --prefix`

WDYT?

Reply:
As per the discussion in #1605, `chezmoi upgrade` is a convenience function for users who have installed chezmoi manually, and I don't think it's realistic or necessary to for it to handle all package manager possibilities here, so I'd prefer to keep this simple.